source/javascripts/components/ProductTour/ProductTour.tsx

```diff
@@ -0,0 +1,97 @@
+/* eslint-disable @typescript-eslint/camelcase */
+import React, { useEffect } from "react";
+import { useCallback, useState } from "react";
+import { Highlighter, useHighlighter } from "./Highlighter";
+import { ProductTooltip } from "./ProductTooltip";
+import { ProductTourProps, Tips } from "./types";
+import { useTrackingFunction } from "./useTrackingFunction";
+import { tips } from "./tips";
+import { useWaitForElements } from "./useWaitForElement";
+
+export const ProductTourContent = ({ menuIds, currentUser }: ProductTourProps): JSX.Element | null => {
+	const [isOpen, setIsOpen] = useState(true);
+	const [validTips, setValidTips] = useState<Tips[] | null>(null);
+	const [selectedId, setSelectedId] = useState("");
+
+	const onFound = useCallback(() => {
+		const filtered = tips.filter(tip => menuIds.includes(tip.id));
+		setValidTips(filtered);
+		setSelectedId(filtered[0].id);
+	}, []);
+
+	// NOTE: Angular passes the menu items ids it will render,
+	// but we stil need to wait for the templates to render them before we can move on.
+	useWaitForElements(menuIds, onFound);
+
+	const { rect, clipPath, highlighted } = useHighlighter(selectedId);
```

Comment:
For me it would be more understandable if `useHighlighter` would be called from the `Highlighter` component, the only thing used here is the `highlighted` property. Wouldn't it be better to have a callback on the `Highlighter` component something like `onHighlighted` and then this hook could be encapsulated within the `Highlighter`

Reply:
Yes I could've went with something like this 
```tsx
<Highlighter>{(rect) => {<ProductTooltip />}}</Highlighter>
```
This would mean the `Highlighter` component fully encapsulates the flow of highlighting. 

But I've opted to slice it in a different way as the only thing they need to share technically is the `DOMRect` data. So technically we could have different combinations of components/hooks for visualising the highlight. I think with me moving out the hook from `Highlighter` makes it more obvious.

Wdyt? Is this acceptable?